qiita_db/metadata_template/base_metadata_template.py

```diff
@@ -1325,3 +1326,78 @@ def check_restrictions(self, restrictions):
                 for col in restriction.columns}
 
         return cols.difference(self.categories())
+
+    def _get_accession_numbers(self, column):
+        """Return the accession numbers stored under the given column
+
+        Parameters
+        ----------
+        column : str
+            The column name where the accession number is stored
+
+        Returns
+        -------
+        dict of {str: str}
+            The accession numbers keyed by sample id
+        """
+        with TRN:
+            sql = """SELECT sample_id, {0}
+                     FROM qiita.{1}
+                     WHERE {2}=%s""".format(column, self._table,
+                                            self._id_column)
+            TRN.add(sql, [self.id])
+            result = {sample_id: accession
+                      for sample_id, accession in TRN.execute_fetchindex()}
+        return result
+
+    def _update_accession_numbers(self, column, values):
```

Comment:
This function is tested through the setters for the ebi accession numbers: `SampleTemplate.ebi_sample_accession`, `SampleTemplate.biosample_accession` and `PrepTemplate.ebi_experiment_accession`. Those 3 functions are calling this one directly and this function cannot be used form the base class (because it cannot be instantiated).